lib/core/error.ts

```diff
@@ -23,3 +23,15 @@ export class ResonateError extends Error {
     return e instanceof ResonateError ? e : new ResonateError(ErrorCodes.UNKNOWN, "Unexpected error", e);
   }
 }
+
+// resonate test error extends ResonateError
+export class ResonateTestError extends ResonateError {
```

Comment:
Can we extend ResonateError and rename this ResonateTestCrash? I think we can hardcode cause to undefined and retryable to false